ui/selectors/transactions.js

```diff
@@ -45,13 +46,27 @@ export const unapprovedEncryptionPublicKeyMsgsSelector = (state) =>
 export const unapprovedTypedMessagesSelector = (state) =>
   state.metamask.unapprovedTypedMessages;
 
+export const smartTransactionsListSelector = (state) =>
+  state.metamask.smartTransactionsState?.smartTransactions?.[
+    getCurrentChainId(state)
+  ]
+    ?.filter((stx) => !stx.confirmed)
+    .map((stx) => ({
+      ...stx,
+      transactionType: 'smart',
+      status: stx.status?.startsWith('cancelled')
```

Comment:
Any reason we can't do an exact match here?

Reply:
Yup, we have multiple statuses with the `cancelled_` prefix, e.g. `cancelled_user_cancelled` or `cancelled_would_revert`.